component-samples/demo/scripts/fdo_toolkit.sh

```diff
@@ -0,0 +1,282 @@
+    echo "Docker is required, installing it..."
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+    chk $? 'adding docker repository key'
+    add-apt-repository "deb [arch=$(dpkg --print-architecture)] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+    chk $? 'adding docker repository'
+    apt-get install -y docker-ce docker-ce-cli containerd.io
+    chk $? 'installing docker'
+fi
+
+# If docker-compose isn't installed, or isn't at least 1.21.0 (when docker-compose.yml version 2.4 was introduced), then install/upgrade it
+# For the dependency on 1.21.0 or greater, see: https://docs.docker.com/compose/release-notes/
+minVersion=2.24.5
+if ! isDockerComposeAtLeast $minVersion; then
+    if [[ -f '/usr/bin/docker-compose' ]]; then
+        echo "Error: Need at least docker-compose $minVersion. A down-level version is currently installed, preventing us from installing the latest version. Uninstall docker-compose and rerun this script."
+        exit 2
+    fi
+#    echo "docker-compose is not installed or not at least version $minVersion, installing/upgrading it..."
+#    # Install docker-compose from its github repo, because that is the only way to get a recent enough version
+#    curl --progress-bar -L "https://github.com/docker/compose/releases/latest/download/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose ## TODO link not working
+#    chk $? 'downloading docker-compose'
+#    chmod +x /usr/local/bin/docker-compose
+#    chk $? 'making docker-compose executable'
+#    ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+#    chk $? 'linking docker-compose to /usr/bin'
+fi
+
+
+echo "Upgraded and Installed all FDO dependencies"
+}
+
+
+update_scripts() {
+cd $1
+sed -i 's/#subjectAltName = @alt_names/subjectAltName = @alt_names/g' web-server.conf
+sed -i 's/#subjectAltName/subjectAltName/g' web-server.conf
+head -n -8 web-server.conf > temp.txt ; mv temp.txt web-server.conf
+
+echo "[ alt_names ]">> web-server.conf
+echo "DNS.1 = $ipaddress" >> web-server.conf
+echo "DNS.2 = localhost" >> web-server.conf
+echo "DNS.3 = host.docker.internal" >> web-server.conf
+echo "DNS.4 = host.docker.internal" >> web-server.conf
+echo "IP.1 = $ipaddress" >> web-server.conf
+echo "IP.2 = 127.0.0.1" >> web-server.conf
+echo "IP.3 = 172.17.0.1" >> web-server.conf
+
+}
+
+generate_keys() {
+cd $1
+bash demo_ca.sh
+bash web_csr_req.sh
+bash user_csr_req.sh
+bash keys_gen.sh
+chmod +r secrets/server-key.pem
+cp -r secrets service.env ../manufacturer/
+cp -r secrets service.env ../rv/
+cp -r secrets service.env ../owner/
+cp -r secrets service.env ../aio/
+cp -r secrets ../db/
+}
+
+start_db() {
+cd $1/db
+if [ ! "$( docker container inspect -f '{{.State.Status}}' $container_name )" == "running" ]; then
+  sed -i 's+innodb_change_buffer_max_size = 25+#innodb_change_buffer_max_size = 25+g' custom/config-file.cnf
+  docker-compose up --build -d
```

Comment:
After executing the command, better to check if the container started properly and running using check


```suggestion
  docker-compose up --build -d
  if [ ! "$( docker container inspect -f '{{.State.Status}}' db_fdo-db_1 )" == "running" ]; then
     echo "Failed to start DB container" >&2
     exit 1
  fi
```